rest-client/src/main/java/org/sagebionetworks/bridge/rest/ApiClientProvider.java

```diff
@@ -41,6 +43,7 @@
     private final Retrofit unauthenticatedRetrofit;
     private final LoadingCache<Class<?>, ?> unauthenticatedServices;
     private final AuthenticationApi authenticationApi;
+    private final SocketFactory socketFactory;
```

Comment:
Need to run some Android specific code

Reply:
This needs to be usable in other Java clients. What if you don't provide a socket factory? Does anything break?

Reply:
Discussed this offline. Added comment explaining the behavior when no factory is provided, and an example use case